pkg/traceql/engine.go

```diff
@@ -231,9 +231,17 @@ func (e *Engine) asTraceSearchMetadata(spanset *Spanset) *tempopb.TraceSearchMet
 		RootTraceName:     spanset.RootSpanName,
 		StartTimeUnixNano: spanset.StartTimeUnixNanos,
 		DurationMs:        uint32(spanset.DurationNanos / 1_000_000),
+		ServiceStats:      make(map[string]*tempopb.ServiceStats, len(spanset.ServiceStats)),
 		SpanSet:           &tempopb.SpanSet{},
 	}
 
+	for service, stats := range spanset.ServiceStats {
```

Comment:
would like to see some of the existing tests extended to cover this and other service stats functionality. this area is pretty well tested and it would be nice to catch this logic as well.

Reply:
Added ServiceStats in the existing TraceQL engine tests: https://github.com/stoewer/tempo/pull/2
Which other areas should be tested? We've coverage of the main logic in `vparquet4/schema.go` and the serviceStatsIters are tested in `block_traceql_meta_test.go`.